src/Raven.Server/Utils/ShardHelper.cs

```diff
@@ -323,5 +338,19 @@ private static void ValidateBucketsMapping(DatabaseRecord record)
                 lastShard = current.ShardNumber;
             }
         }
+
+        public static string GenerateStickyId(string id, int bucket, char identityPartsSeparator)
+        {
+            Debug.Assert(id[^1] == identityPartsSeparator, "id[^1] == identityPartsSeparator");
+
+            return $"{id}${bucket}{SuffixIdTerminator}";
```

Comment:
so the hash will be generated based on `{bucket}{SuffixIdTerminator}` ?

Reply:
Sorry it is a bit hard for me to follow the end result here
From the orchestrator we generate something like `Users/$123*` and send this the shard which is the result of hashing only the `123`, right?
But on the shard itself we end up with what kind of id?
if it is `users/00000565-A$123`, then we okay.